src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -130,11 +138,23 @@ public synchronized void updateSubscriptions(Set<String> topics, @NonNull Consum
 
         topicsToSubscribe.forEach(s -> {
             try {
-                subscribeToIotCore(s);
-                LOGGER.atDebug().kv(TOPIC, s).log("Subscribed to topic");
-                subscribedIotCoreTopics.add(s);
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
-                LOGGER.atError().kv(TOPIC, s).log("Failed to subscribe");
+                RetryUtils.runWithRetry(subscribeRetryConfig, () -> {
+                    try {
+                        subscribeToIotCore(s);
+                        subscribedIotCoreTopics.add(s);
+                        // useless return
+                        return null;
+                    } catch (ExecutionException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)